packages/integration/src/lib/workspace/workspace-button/workspace-button.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, Input, ChangeDetectionStrategy, OnInit, OnDestroy } from '@angular/core';
+import { Layer, VectorLayer } from '@igo2/geo';
+import { WorkspaceState } from '../workspace.state';
+import { BehaviorSubject, Subscription } from 'rxjs';
+
+@Component({
+  selector: 'igo-workspace-button',
+  templateUrl: './workspace-button.component.html',
+  styleUrls: ['./workspace-button.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class WorkspaceButtonComponent implements OnInit, OnDestroy {
+
+  public hasWorkspace$: BehaviorSubject<boolean> = new BehaviorSubject(false);
+  private hasWorkspace$$: Subscription;
+
+  @Input() layer: Layer;
+
+  @Input() color: string = 'primary';
+
+  constructor(private workspaceState: WorkspaceState) {}
+
+  ngOnInit(): void {
+    this.hasWorkspace$$  = this.workspaceState.workspaceEnabled$.subscribe(wksEnabled =>
+      (wksEnabled && this.layer instanceof VectorLayer) ? this.hasWorkspace$.next(true) : this.hasWorkspace$.next(false)
```

Comment:
You could replace this long line with:` this.hasWorkspace$.next(wksEnabled && this.layer instanceof VectorLayer)`

Reply:
I still think it is a subcribe that I need.

Reply:
Yes, the subscribe is still needed. My suggestion was to replace the ternary operator.

Reply:
fixed